packages/react/src/transactions/infrastructure/__tests__/createPublicationMetadata.spec.ts

```diff
@@ -121,36 +126,65 @@ describe(`Given the ${createPublicationMetadata.name} helper`, () => {
         });
       });
 
-      describe(`without appId`, () => {
+      describe(`with media`, () => {
+        it(`should return the metadata.media with only defined fields`, () => {
+          const request = mockPublication({
+            media: [mockMediaObject()],
+            locale: 'en',
+            contentFocus: ContentFocus.IMAGE,
+          });
+
+          const metadata = createPublicationMetadata(request);
+
+          expect(metadata.media?.[0]).not.toHaveProperty('altTag');
+          expect(metadata.media?.[0]).not.toHaveProperty('cover');
+        });
+      });
+
+      describe(`without optional fields like appId, animationUrl etc.`, () => {
         it(`should return the metadata without appId field`, () => {
           const request = mockPublication({
             content,
-            media,
             locale: 'en',
             contentFocus: ContentFocus.TEXT,
-            collect: mockNoCollectPolicy(),
           });
 
           const metadata = createPublicationMetadata(request);
 
+          expect(metadata).not.toHaveProperty('animation_url');
           expect(metadata).not.toHaveProperty('appId');
+          expect(metadata).not.toHaveProperty('contentWarning');
+          expect(metadata).not.toHaveProperty('external_url');
+          expect(metadata).not.toHaveProperty('image');
+          expect(metadata).not.toHaveProperty('imageMimeType');
+          expect(metadata).not.toHaveProperty('tags');
         });
       });
 
-      describe(`with media`, () => {
+      describe(`with all optional fields like appId, animationUrl etc.`, () => {
         it(`should return the metadata.media with only defined fields`, () => {
           const request = mockPublication({
             content,
-            media: [mockMediaObject()],
             locale: 'en',
             contentFocus: ContentFocus.TEXT,
-            collect: mockNoCollectPolicy(),
+            animationUrl: faker.internet.url(),
+            appId: appId('test'),
+            contentWarning: ContentWarning.NSFW,
+            externalUrl: faker.internet.url(),
+            image: faker.image.imageUrl(),
+            imageMimeType: ImageType.JPEG,
+            tags: [faker.lorem.word()],
           });
 
           const metadata = createPublicationMetadata(request);
 
-          expect(metadata.media?.[0]).not.toHaveProperty('altTag');
-          expect(metadata.media?.[0]).not.toHaveProperty('cover');
+          expect(metadata.animation_url).toBeDefined();
+          expect(metadata.appId).toBeDefined();
+          expect(metadata.contentWarning).toEqual('NSFW');
+          expect(metadata.external_url).toBeDefined();
+          expect(metadata.image).toBeDefined();
+          expect(metadata.imageMimeType).toBeDefined();
+          expect(metadata.tags).toBeDefined();
```

Comment:
I think we need to support `cover` as well. I would assume this is just a URL of the original image (like for `media.cover` and `media.url`). I asked cause it's not documented in the docs.lens.xyz.

Reply:
Ok, the origin of this seems not clear, but it's basically the `image` field exposed as `MediaSet` (which currently is not that more useful) but with `optimized` image URL being added to `MediaSet` it can become useful.

So there is technically nothing to do on the write path.

Reply:
afaik the `cover` is not even mentioned in the metadata specs, I didn't test it but I guess it would not pass the verification.